### PR TITLE
Native manifest

### PR DESCRIPTION
**Pull request by deitch**
**- What I did**

* Remove the dependency on an external [manifest-tool](https://github.com/estesp/manifest-tool) by using it as a library
* updated a lot of dependencies and the long tree (web?) it takes to get it there

**- How I did it**

painfully

**- How to verify it**

run it

**- Description for the changelog**

no longer need external manifest-tool

**Notes**

* I had to update a _lot_ of dependencies in `src/cmd/linuxkit/vendor/` based on `vendor.conf`. I need some real reviews on this.
* I did it in 3 commits to make it easier to review. Once we are satisfied, I will squash them into 1
* this only works with go 1.14 or higher, so our go-compile package will fail on this. I will get a separate PR in to update that

### CODE REVIEW

File: pkg/init/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,5 @@
 FROM linuxkit/alpine:bc528cf9d4065d2e09aa44ff76909b94cfe8d867 AS build
+
```
Comment (rn) [inline pkg/init/Dockerfile:2]: seems like an unrelated/accidental change.

Comment (deitch) [reply pkg/init/Dockerfile:2]: I have not the foggiest clue how this got in here. Pretty funny. Fixing now. Thanks for catching @rn

Comment (deitch) [reply pkg/init/Dockerfile:2]: Fixed.

**Comment by rn** [comment]
I modified `pkg/format` and tried to push it with the modified `linuxkit` and then pull with DCT enabled. This seems to work on x86 but fails on arm64:

```
$ DOCKER_CONTENT_TRUST_REPOSITORY_PASSPHRASE=XXXXX ../../bin/linuxkit pkg push .
[...]
Pushing linuxkit/format:d0a740f11aad799aa5125e75dbb68668c4b49a8d-arm64
The push refers to repository [docker.io/linuxkit/format]
71f96de3671e: Pushed
d54a6443c57f: Pushed
d0a740f11aad799aa5125e75dbb68668c4b49a8d-arm64: digest: sha256:22188ef178a67cedfc8943ac4a03ed29946aaf4d3a02439ce8836130cdaa39d3 size: 738
Signing and pushing trust metadata
[...]
Successfully signed docker.io/linuxkit/format:d0a740f11aad799aa5125e75dbb68668c4b49a8d-arm64
Pushing linuxkit/format:d0a740f11aad799aa5125e75dbb68668c4b49a8d-arm64 to manifest linuxkit/format:d0a740f11aad799aa5125e75dbb68668c4b49a8d
Retrieving digests of images...
WARN[0049] Couldn't find or access image reference "linuxkit/format:d0a740f11aad799aa5125e75dbb68668c4b49a8d-amd64". Skipping image.
Image "linuxkit/format:d0a740f11aad799aa5125e75dbb68668c4b49a8d-arm64" is digest sha256:22188ef178a67cedfc8943ac4a03ed29946aaf4d3a02439ce8836130cdaa39d3; size: 738
WARN[0049] Couldn't find or access image reference "linuxkit/format:d0a740f11aad799aa5125e75dbb68668c4b49a8d-s390x". Skipping image.
Signing manifest for linuxkit/format:d0a740f11aad799aa5125e75dbb68668c4b49a8d
notary works with sha256 hash, not the provided linuxkit/format

$ DOCKER_CONTENT_TRUST=1 docker pull linuxkit/format:d0a740f11aad799aa5125e75dbb68668c4b49a8d-arm64
Pull (1 of 1): linuxkit/format:d0a740f11aad799aa5125e75dbb68668c4b49a8d-arm64@sha256:22188ef178a67cedfc8943ac4a03ed29946aaf4d3a02439ce8836130cdaa39d3
sha256:22188ef178a67cedfc8943ac4a03ed29946aaf4d3a02439ce8836130cdaa39d3: Pulling from linuxkit/format
Digest: sha256:22188ef178a67cedfc8943ac4a03ed29946aaf4d3a02439ce8836130cdaa39d3
Status: Image is up to date for linuxkit/format@sha256:22188ef178a67cedfc8943ac4a03ed29946aaf4d3a02439ce8836130cdaa39d3
Tagging linuxkit/format@sha256:22188ef178a67cedfc8943ac4a03ed29946aaf4d3a02439ce8836130cdaa39d3 as linuxkit/format:d0a740f11aad799aa5125e75dbb68668c4b49a8d-arm64
docker.io/linuxkit/format:d0a740f11aad799aa5125e75dbb68668c4b49a8d-arm64

$ DOCKER_CONTENT_TRUST=1 docker pull linuxkit/format:d0a740f11aad799aa5125e75dbb68668c4b49a8d
No valid trust data for d0a740f11aad799aa5125e75dbb68668c4b49a8d 
```

So, the DCT pull of the image worked but the DCT pull of the manifest list failed because it was not signed.

**Comment by deitch** [comment]
> Signing manifest for linuxkit/format:d0a740f11aad799aa5125e75dbb68668c4b49a8d
notary works with sha256 hash, not the provided linuxkit/format

That is the key. Missed something there. Going back to check

**Comment by deitch** [comment]
Try now.

**Comment by rn** [comment]
did you test it?

**Comment by deitch** [comment]
Ha! Not yet. I found the fix and ran the unit tests, but updated right away. Trying now.

**Comment by deitch** [comment]
That looks a lot better:

```
DEBU[0034] mediaType of manifestList: application/vnd.docker.distribution.manifest.list.v2+json
Signing manifest for linuxkit/format:8b89b05b9ccd614a02d3be57e86428030d7482f8
DEBU[0037] Executing: [notary -s https://notary.docker.io -d /Users/adeitcher/.docker/trust addhash -p docker.io/linuxkit/format 8b89b05b9ccd614a02d3be57e86428030d7482f8 433 --sha256 5c74bb4396c38c1f95002be312aabe4d3e5595bb36a3cb006512df2ca1251e1c -r targets/releases]
Addition of target "8b89b05b9ccd614a02d3be57e86428030d7482f8" by sha256 hash to repository "docker.io/linuxkit/format" staged for next publish.
Auto-publishing changes to docker.io/linuxkit/format
WARN[0006] targets/justin metadata is nearing expiry, you should re-sign the role metadata
WARN[0006] targets/riyaz metadata is nearing expiry, you should re-sign the role metadata
WARN[0006] targets/ian metadata is nearing expiry, you should re-sign the role metadata
WARN[0006] targets metadata is nearing expiry, you should re-sign the role metadata
Successfully published changes for repository docker.io/linuxkit/format
New signed multi-arch image: linuxkit/format:8b89b05b9ccd614a02d3be57e86428030d7482f8
Build and push complete, not releasing, all done.
```

**Comment by deitch** [comment]
Yes, looks good, at least on amd64. I built+pushed+signed from one host, then did `DOCKER_CONTENT_TRUST=1 docker pull .... `

Going to arm64 next

**Comment by deitch** [comment]
Works like a charm @rn . Built+pushed+signed on amd64, repeated on arm64, removed all local images on amd64, and then did `DOCKER_CONTENT_TRUST=1 docker pull ...` and it worked.

**Comment by deitch** [comment]
I also finally squashed the commits :-)
